core-data/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress Dependencies
+ */
+import {
+	registerReducer,
+	registerSelectors,
+	registerResolvers,
+} from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import reducer from './reducer';
+import { getCategories } from './selectors';
+
+/**
+ * Module Constants
+ */
+const MODULE_KEY = 'core';
+
+const store = registerReducer( MODULE_KEY, reducer );
+
+registerSelectors( MODULE_KEY, { getCategories } );
+registerResolvers( MODULE_KEY, {
+	getCategories: {
+		fulfill: () => {
```

Comment:
Minor: Function property shorthand is more compact than arrow function:

```js
fulfill: () => {
fulfill() {
```

Reply:
Does `fulfill` need to be a property of the object? Are we anticipating more properties? Couldn't the function just be the value of the `getCategories` key?

Reply:
> Are we anticipating more properties?

I thought about this and I think we should keep it as an object to allow extra properties in the future. I can think of a `TTL` property for example.

Reply:
There a few alternative ways to do it if we need to:
* function when no additional properties or object otherwise - type detection in the background
* higher-order function which adds additional behavior
* return value from the `fulfill` implementation

Just saying that we don't need to default to the object. I'd be in favor of starting with the function and change it if we find it too limited.

Reply:
Yeah, I think setting it as a function is still forward-compatible with the patterns @gziolo outlined, and a nice simple usage when you don't need more complex options.

```js
function normalizeResolver( resolver ) {
	if ( typeof resolver === 'function' ) {
		return { fulfill: resolver };
	}

	return resolver;
}
```